airflow/contrib/hooks/gcp_automl_hook.py

```diff
@@ -0,0 +1,727 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""
+This module contains a Google AutoML hook.
+"""
+from typing import Dict, Optional, Sequence, Tuple, Union, List
+
+from google.api_core.retry import Retry
+from google.api_core.gapic_v1.client_info import ClientInfo
+from google.cloud.automl_v1beta1 import PredictionServiceClient, AutoMlClient
+from google.cloud.automl_v1beta1.types import (
+    BatchPredictInputConfig,
+    BatchPredictOutputConfig,
+    Model,
+    ExamplePayload,
+    Dataset,
+    FieldMask,
+    ImageObjectDetectionModelDeploymentMetadata,
+    PredictResponse,
+    ColumnSpec,
+    Operation,
+    TableSpec,
+    InputConfig,
+)
+
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from airflow.version import version
+
+
+def get_object_id(obj: dict) -> str:
+    """
+    Returns unique id of the object.
+    """
+    return obj["name"].rpartition("/")[-1]
+
+
+class AutoMLHook(GoogleCloudBaseHook):
+    """
+    Google Cloud AutoML hook.
+
+    All the methods in the hook where project_id is used must be called with
+    keyword arguments rather than positional.
+    """
+
+    def __init__(self, gcp_conn_id: str = "google_cloud_default", delegate_to=None):
+        super().__init__(gcp_conn_id, delegate_to)
+        self._client_info = ClientInfo(client_library_version="airflow_v" + version)
```

Comment:
Please unify to https://github.com/apache/airflow/commit/8dcb9763771957017a2e6fe9db6f67a068463774